core/dbt/cli/requires.py

```diff
@@ -346,15 +370,22 @@ def setup_manifest(ctx: Context, write: bool = True, write_perf_info: bool = Fal
 
     runtime_config = ctx.obj["runtime_config"]
 
+    catalogs = ctx.obj["catalogs"] if "catalogs" in ctx.obj else []
+    active_write_integrations = [get_active_write_integration(catalog) for catalog in catalogs]
```

Comment:
Shouldn't there only be one (or none) active write integrations?

Reply:
Per catalog, yes. But there may be more than 1 catalog like so:

```yaml
catalogs:
  - name: catalog1
    ...
  - name: catalog2
    ...
  ...
```